include/sys/types.h

```diff
@@ -121,20 +121,27 @@ typedef uint16_t     size_t;
 typedef int16_t      ssize_t;
 typedef uint16_t     rsize_t;
 
+/* uid_t is used for user IDs
+ * gid_t is used for group IDs.
+ */
+
+typedef int16_t      uid_t;
+typedef int16_t      gid_t;
+
 #else /* CONFIG_SMALL_MEMORY */
 
 typedef _size_t      size_t;
 typedef _ssize_t     ssize_t;
 typedef _size_t      rsize_t;
 
-#endif /* CONFIG_SMALL_MEMORY */
-
 /* uid_t is used for user IDs
  * gid_t is used for group IDs.
  */
 
-typedef int16_t      uid_t;
-typedef int16_t      gid_t;
+typedef unsigned int uid_t;
```

Comment:
My only concern is usage of non-fixed length type `unsigned int` vs `uint32_t`.

Reply:
There are other types using `unsigned int` in this file, using `unsigned int` just follow linux:
https://github.com/torvalds/linux/blob/master/include/linux/types.h#L37-L38
https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/posix_types.h#L36-L37